src/main/java/com/ebay/myriad/scheduler/event/handlers/ResourceOffersEventHandler.java

```diff
@@ -112,10 +117,7 @@ public void onEvent(ResourceOffersEvent event, long sequence,
                     }
                 }
             } else {
-                LOGGER.info("No pending tasks, declining all offers");
-                for (Offer offer : offers) {
-                    driver.declineOffer(offer.getId());
-                }
+                offerLifecycleMgr.addOffers(offers);
             }
```

Comment:
Great point. Instead of rejecting the offer, we should simply add it to the offer feed using offerLifecycleMgr.addOffers. This is a regression because earlier without fine grained scaling, we would always reject the offer since the NM was launched with pre configured capacity. We forgot to update this class for fine grained scheduling.